pkg/inject/inject.go

```diff
@@ -539,6 +546,32 @@ func (conf *ResourceConfig) serviceAccountVolumeMount() *corev1.VolumeMount {
 	return nil
 }
 
+func (conf *ResourceConfig) trace() *charts.Trace {
+	var (
+		svcAddr    = conf.getOverride(k8s.ProxyTraceCollectorSvcAddr)
+		svcAccount = conf.getOverride(k8s.ProxyTraceCollectorSvcAccount)
+	)
+
+	if svcAddr == "" {
+		return nil
+	}
+
+	splits := strings.Split(svcAddr, ".")
+	if len(splits) <= 1 {
+		log.Errorf("fail to enable tracing due to malformed collector service address %s. Expected service address to use the <svc-name>:<ns> format", svcAddr)
+		return nil
+	}
+
+	if delimiter := strings.Index(splits[1], ":"); delimiter != -1 {
```

Comment:
What is this doing?  Stripping the port off?  Is there a more clear way to do this, perhaps before splitting the domain into segments?